ts/features/fci/utils/signatureFields.ts

```diff
@@ -80,3 +81,23 @@ export const getSectionListData = (
       data: clausesByType(signatureFields, [type])
     }))
   );
+
+/**
+ * Orders the signatureFields array with the given order: UNFAIR -> REQURED -> EVERYTHING ELSE (OPTIONAL)
+ * @param signatureFields an array of signature fields
+ * @returns the new ordered array
+ */
+export const orderSignatureFields = (
+  signatureFields: ReadonlyArray<SignatureField>
+): ReadonlyArray<SignatureField> =>
+  pipe(
+    _.orderBy(signatureFields, [
+      field =>
+        field.clause.type === ClauseTypeEnum.UNFAIR
+          ? 0
+          : field.clause.type === ClauseTypeEnum.REQUIRED
+          ? 1
+          : 2
+    ]),
+    RA.fromArray
+  );
```

Comment:
I personally prefer to [avoid using of lodash](https://medium.com/techspiration/why-you-shouldnt-use-lodash-f8504d7b7383) if we have [all we need](https://dev.to/gcanti/getting-started-with-fp-ts-ord-5f1e) from fp-ts.

What do you think about using contramap and sortBy from fp-ts?

This could be a possibile implementation:
```typescript
const byClausesType = pipe(
  N.Ord,
  contramap((signatureField: SignatureField) => {
    switch (signatureField.clause.type) {
      case ClauseTypeEnum.UNFAIR:
        return 0;
      case ClauseTypeEnum.REQUIRED:
        return 1;
      default:
        return 2;
    }
  })
);

const sortByType = RA.sortBy([byClausesType]);

export const orderSignatureFields = (
  signatureFields: ReadonlyArray<SignatureField>
): ReadonlyArray<SignatureField> => pipe(signatureFields, sortByType);

```

Reply:
Classy solution, thank you. I implemented a minor variation with an explicit `OPTIONAL` case and added a few tests as suggested.